src/bundle/bundle.ts

```diff
@@ -0,0 +1,64 @@
+import esbuild from "esbuild"
+import { createRouteMapFromDirectory } from "src/routes/create-route-map-from-directory";
+
+const alphabet = "zyxwvutsrqponmlkjihgfedcba"
+
+const getRandomId = (length: number): string => {
+  let str = '';
+  let num = length;
+  while (num--) str += alphabet[Math.random() * alphabet.length | 0];
+  return str;
+}
+
+interface BundleOptions {
+  directoryPath: string,
+  /**
+   * This should not be provided in most cases so your bundle is maximally portable.
+   */
+  bundledAdapter?: "wintercg-minimal"
+}
+
+export const bundle = async (options: BundleOptions) => {
```

Comment:
the idea is that this will more or less be directly wrapped in a `npx edgespec bundle` command